### ISSUE TITLE
Tabset ul not containing li's according to lighthouse

### ISSUE THREAD

**Issue by internalsystemerror** [ng-bootstrap/ng-bootstrap#3134]
Current behavior
~~When using the tabset, the ul has a role of tabset and the a elements within the li's have a role of tab.~~
This has now changed to say the errors reported by @IAfanasov, that li's are not contained within a ul, even though they are.

~~Expected behavior
No roles are assigned to lists.

This is incorrect Aria behaviour and fails a Google Lighthouse test. See documentation: https://getbootstrap.com/docs/4.3/components/navs/#regarding-accessibility as to why this should be removed.

This was introduced by the following issue (however well intentioned) https://github.com/ng-bootstrap/ng-bootstrap/issues/1118. All I know is, this is hindering those perfect lighthouse scores we're all going for under accessibility :D.~~
The problem does seem to be specific to using ng-bootstrap's tablist and doesn't occur when typing the HTML out by hand, possibly due to the UL and LIs being part of different components?

**Comment by IAfanasovMob** [comment]
Hi @internalsystemerror ! (:

I run the demo locally with command `yarn demo`. Opened the tabset page, run the lighthouse audit and don't see this issue. May you please point me on it so I can test and fix it? (: 

<img width="1680" alt="Screen Shot 2019-06-05 at 20 58 23" src="https://user-images.githubusercontent.com/44999963/58982417-b1f5c000-87d4-11e9-823f-42b3dff5f764.png">

even updated chrome and still can't see the specified error (:

**Comment by internalsystemerror** [comment]
Tested again and lighthouse no longer reports this. Closing as not an issue.

**Comment by internalsystemerror** [comment]
On closer inspection the error shown in your list "List items (<li>) are not contained within <ul> or <ol> parent elements." is the same one I'm now getting in lighthouse for this tablist. I will check whether this issue is correctly described.

**Comment by iamgraviteja** [comment]
@internalsystemerror Yes. I observed the same issue even though the `<li>` elements appear in the `<ul>`. Could you please let me know when you find anything on this?

**Comment by internalsystemerror** [comment]
I've updated the issue to reflect the current status, and although it is specific to ng-bootstrap, I'm not sure what ng-bootstrap can do to resolve it, and I would suggest that it's an error with lighthouse to not correctly parse the components? The HTML on manual inspection appears to be correct.
![image](https://user-images.githubusercontent.com/1626298/61118193-1836c800-a490-11e9-9713-800ae8f87a1c.png)
![image](https://user-images.githubusercontent.com/1626298/61118231-2c7ac500-a490-11e9-815f-558058768aa5.png)
![image](https://user-images.githubusercontent.com/1626298/61118259-40bec200-a490-11e9-9a01-bc9b8958c6ae.png)

I think it's something we may have to just put up with. Would be interested to hear any other point of views on this.

**Comment by iamgraviteja** [comment]
@internalsystemerror It's not only with the lighthouse but I observed using [Accessibility Insights for web](https://chrome.google.com/webstore/detail/accessibility-insights-fo/pbjjkligggfmakdaogkfomddhfmpjeni?hl=en).

**Comment by constantincechina** [comment]
The issue is still present, even in latest sample from the example page.
Although markup is from HTML correct, having the role=tablist, is what confuses the screen readers.

Issue can be easily identified with Accesibility from DevTools, [AXE](https://www.deque.com/axe/) [or Accessibility Insights for Web](https://accessibilityinsights.io/docs/en/web/overview).

![image](https://user-images.githubusercontent.com/17467157/62827627-a9d54a80-bba0-11e9-93c7-8dab7bd1b827.png)

The fix seems to be changing the role="tablist" to role="list" or removing it altogether.

A great explanation in [this article](http://simplyaccessible.com/article/danger-aria-tabs/) explain the issue from real users perspective. their solution includes the removal of the role="tablist".

**Comment by sysmat** [comment]
Any progress on this exe problem because Impact is serious?

Regards

**Comment by maxokorokov** [comment]
At the moment we were following Bootsrap's markup completely.
There is also `NgbNav` in works here → #3492

It should at least be possible to override `role` like so:

```html
<ul ngbNav class="nav-tabs" [attr.role]="...">
```

**Comment by internalsystemerror** [comment]
The markup is not being followed as stated here: https://getbootstrap.com/docs/4.3/components/navs/#regarding-accessibility the role should be set as navigation not tablist.

**Comment by maxokorokov** [comment]
@internalsystemerror `NgbTabset` is not intended to be used in navbars, so it should be `tablist` → https://getbootstrap.com/docs/4.3/components/navs/#javascript-behavior. 

On the other hand I think it might be a good idea to add an opt-out option for setting roles in #3492, even though it is possible to override them

**Comment by internalsystemerror** [comment]
Ah, I read it as a general navigation bar, not a "navbar". Are the other roles set to tab and tab panel accordingly? If so, are Google the ones who are incorrect here?

**Comment by maxokorokov** [comment]
Since version `5.2.0` there is a new `NgbNav` set of directives that fix this issue. `NgbNav` is meant to replace the `NgbTabset` component (completely deprecated in version `6.0.0`) as a more flexible alternative.

There is also a new `[roles]` input in the `NgbNav` → https://ng-bootstrap.github.io/#/components/nav/api#NgbNav

I suggest you migrate to it → https://ng-bootstrap.github.io/#/components/nav/overview
